src/ast/break-statement.ts

```diff
@@ -0,0 +1,20 @@
+import Context from '../semantics/context';
+import { IAstNode } from '../type-definitions/plainscript-types';
+
+export default class BreakStatement implements IAstNode<BreakStatement> {
+  // no constructor on purpose.
+
+  public analyze(context: Context): void {
+    if (!context.inLoop) {
+      throw new Error('Break statement outside loop');
+    }
+  }
+
+  public optimize(): BreakStatement {
+    return this;
+  }
+
+  // Depends on the target language, thus gets filled in
+  // by the necessary generator at runtime.
+  public gen() { }
```

Comment:
So `gen` has to be present bc TypeScript?

Reply:
This is discussed in a comment down below, but for posterity: yes indeed ! The generator later down the line gets a compiler error when we try to assign to the `gen` properties saying that the `gen` property does not exist if we do not define it somewhere beforehand ! But as discussed these properties are being consolidated in an `AstNode` base class